### PR TITLE
Ch5 - Splash Screen and Input management

### PR DESCRIPTION
**Pull request by jelster**
This has been a bit of a slog to get here, and it's still not there yet. 

Issues that have to be addressed before this can be merged:

- [x] Re-entrancy, or unintended repeated invocations of the same action. Pressing a key fires about a half-dozen events, but only a single game action (per some period of time) should be allowed. Looking into adding a debounce time threshold
- [x] Scene controls attaching and detaching needs validation that it is not causing extra, "phantom" or other types of eventing issues
- [x] Proper removal of Observers from Observables not automatically handled at dispose time (e.g., scene.dispose)
- [x] Main menu item selection using keyboard/gamepad only partially works

Issues that would hopefully be addressed but aren't blockers (will need to be resolved later):

~~- [ ] Sounds/music management between screens/scenes - audio engine is global, but music started in a scene may get orphaned ~~ moved to #48 
- [x] Gamepad support testing and validated - currently blocked by issue with invalid gamepad devices causing exceptions in BJS input system (5.0.0-alpha-22-preview)
- [x] Main menu exit command needs to signal application to transition to exiting state
- [x] SpaceTruckerApplication needs to properly prepare for exiting by detaching from canvas, engine events

### CODE REVIEW

**Comment by jelster** [comment]
Re-entrancy "solved" by adding a bounce functor that adds a (currently) 250ms debounce to any incoming input action. I don't like that it feels like this is more of a band-aid than a real solution but I think it will be sufficient for now

**Comment by jelster** [comment]
Okay -- just did a fresh clone, `npm i` and then `npm run start` -- ran just fine. Have you tried deleting your package.lock.json then running `npm ci`?

**Comment by jelster** [comment]
> I use 14.15 stable.
Okay -- I'm on node v16.2.0, I just did a fresh clone, npm i and then npm run start -- ran just fine. Have you tried deleting your package.lock.json then running npm ci? It's possible the updated HtmlWebPackPlugin doesn't support node v14, but that's just a guess

**Review by RaananW — COMMENTED** [review]
Late review (sorry for that)

TL;dr : LGTM! :-)


There are a few issues with var/let/const usage (maybe running eslint during development will keep those issues out of the dev). 
Babylon-wise, if it works - it works. I understand the need to abstract input support, and this looks like a great solution for that.
